src/MSALWrapper/AuthFlow/IntegratedWindowsAuthentication.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.AuthFlow
+{
+#if NET472
+    using Microsoft.Identity.Client.Desktop;
+#endif
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Identity.Client;
+
+    /// <summary>
+    /// The broker auth flow.
+    /// </summary>
+    public class IntegratedWindowsAuthentication : IAuthFlow
+    {
+        private readonly ILogger logger;
+        private readonly IEnumerable<string> scopes;
+        private readonly string preferredDomain;
+        private readonly IList<Exception> errors;
+        private IPCAWrapper pcaWrapper;
+
+        #region Public configurable properties
+
+        /// <summary>
+        /// The silent auth timeout.
+        /// </summary>
+        private TimeSpan silentAuthTimeout = TimeSpan.FromMinutes(5);

```

Comment:
Very good point. I tried to stay consistent with the timeout we set for silent auth in Broker.
@kyle-rader - Ideally, silent auth should be in about 300 ms as I heard in our demo today. So should we change the timeout limit to 5 seconds(more than ample time) in future PRs?

Reply:
If an auth flow is doing a "silent auth token refresh" the web requests _can_ take as much as 5 seconds. IWA doesn't do this, though still depends on network traffic. But as @Penguinwizzard mentions - 5 minutes is, too generous. 6 seconds might be a more reasonable timeout for silent auth. @shalinikhare27 I think you can go ahead and change this to 5 seconds here, because IWA can only ever be silent, or fail, and if it takes more than 5 seconds, we probably want to move on.

Reply:
Update the timeout to be 6 seconds.